force-app/main/default/classes/RD2_SustainerEvaluationService.cls

```diff
@@ -114,10 +120,27 @@ public inherited sharing class RD2_SustainerEvaluationService {
         List<SObject> lapsedSustainerCandidates = queryService.getSObjectWithLapsedRecurringDonations(sObjectType, UTIL_SObject.extractIds(inactiveSustainers));
         List<SObject> formerSustainers = evaluateCurrentSustainerTypeAndReturnNotMatchingRecords(SustainerType.Lapsed, lapsedSustainerCandidates);
 
-        for (SObject formerSustainer : formerSustainers) {
-            if (formerSustainer.get(SUSTAINER_FIELD) != SustainerType.Former.name()) {
-                formerSustainer.put(SUSTAINER_FIELD,SustainerType.Former.name());
-                this.reocrdsToUpdate.add(formerSustainer); 
+        if (formerSustainers.size() == 0) {
+            return;
+        }
+  
+        if (isBatchMode) {
```

Comment:
I haven't gone through this super closely, but how are we handling Former sustainers in the Trigger context?

Reply:
I think I found it: `evaluateCurrentSustainerTypeAndReturnNotMatchingRecords`?

Reply:
for trigger context, we need to query back Closed RD and evaluate down to null Sustainer. `List<SObject> nonSustainers = evaluateCurrentSustainerTypeAndReturnNotMatchingRecords(SustainerType.Former, formerSustainerCandidates);` in the next line update the Sustainer type to Former and return all the left over Contact/Account which will be updated to Null.

Reply:
What you think of changing evaluateCurrentSustainerTypeAndReturnNotMatchingRecords to evaluateCurrentSustainerTypeAndReturnUnmatched??